precompiles/erc20/query.go

```diff
@@ -20,3 +36,199 @@ const (
 	// query.
 	BalanceOfMethod = "balanceOf"
 )
+
+// Name returns the name of the token. If the token metadata is registered in the
+// bank module, it returns its name. Otherwise it returns the base denomination of
+// the token capitalized (eg. uatom -> Atom).
+func (p Precompile) Name(
```

Comment:
Should this be pointer semantics instead?

Reply:
why?